rasa/core/lock_store.py

```diff
@@ -223,19 +226,32 @@ def __init__(
             ssl=use_ssl,
             socket_timeout=socket_timeout,
         )
+
+        if (prefix is None) or (prefix == ""):
```

Comment:
this logic should probably be its own method (`_set_prefix(self, prefix)` or something similar) with its own unit tests

Reply:
Hi @ricwo are you thinking something like this?: 

`
async def test_redis_lock_store_with_prefixes(monkeypatch: MonkeyPatch):
    import redis.exceptions

    lock_store = FakeRedisLockStore()
    assert lock_store._set_redis_prefix("!asdf234 34#") == False
    
    monkeypatch.setattr(
        lock_store,
        lock_store.get_or_create_lock.__name__,
        Mock(side_effect=redis.exceptions.TimeoutError),
    )

    with pytest.raises(LockError):
        async with lock_store.lock("some sender"):
            pass

    lock_store = FakeRedisLockStore()
    assert lock_store._set_redis_prefix("prefix") == True
    
    monkeypatch.setattr(
        lock_store,
        lock_store.get_or_create_lock.__name__,
        Mock(side_effect=redis.exceptions.TimeoutError),
    )  

    with pytest.raises(LockError):
        async with lock_store.lock("some new sender"):
            pass
`

Reply:
this would be clearer if a method `_get_redis_prefix()` would return the actual prefix, so that in the constructor of the redis tracker/lock stores we can have
```python
self.prefix = self._get_redix_prefix()
```

Reply:
in addition, I think the attribute should be called something else to make it clear what the prefix pertains to. something like
```python
self.key_prefix = ...
```